keps/sig-api-machinery/20190426-crd-defaulting.md

```diff
@@ -285,7 +304,7 @@ The `default` field in the CRD types is considered alpha quality. We will add a
 
 **blockers for beta:**
 
-* no new tests
+* add tests for default value validation.
```

Comment:
add details of test coverage?
* CRD schema with default containing unknown field inside metadata
  * new CRD (allowed)
  * updated CRD where existing CRD did contain the unknown field (allowed)
  * updated CRD where existing CRD did not contain the unknown field (allowed)
  * building CR storage from persisted CRD discards unknown fields
* CRD schema with default containing schema-invalid field inside metadata (e.g. `finalizers: 1`)
  * new CRD (forbidden)
  * updated CRD where existing CRD did contain the invalid field (allowed)
  * updated CRD where existing CRD did not contain the invalid field (forbidden)
  * building CR storage from persisted CRD discards schema invalid fields from defaults

Reply:
added